libr/search/search.c

```diff
@@ -415,20 +413,8 @@ R_API int r_search_kw_add(RSearch *s, RSearchKeyword *kw) {
 	return true;
 }
 
-R_API void r_search_kw_reset(RSearch *s) {
-	r_list_free (s->kws);
-	s->kws = r_list_new ();
-}
-
 R_API void r_search_reset(RSearch *s, int mode) {
-	R_FREE (s->data);
-	r_list_purge (s->hits);
 	s->nhits = 0;
-	s->hits = r_list_newf ((RListFree)free);
-	if (!s->hits) {
-		return;
-	}
-	r_search_kw_reset (s);
```

Comment:
too much removed code here

Reply:
Destroy these data structures after the current search.